Framework/Core/include/Framework/DataAllocator.h

```diff
@@ -328,6 +344,80 @@ class DataAllocator
   RootObjectContext *mRootContext;
 };
 
+// These are free functions which depend on a global DataAllocator.
+template <typename T>
+typename std::enable_if<is_messageable<T>::value == true, T&>::type
+make(const OutputSpec& spec) {
+  auto &allocator = DataAllocator::getInstance();
+  return allocator.make<T>(spec);
+}
```

Comment:
Maybe we could have only one function with variable (template)arguments forwarding to `DataAllocator` methods. This would avoid the code duplication.